### PR TITLE
[MM-18193] Use vendor folder when compiling test plugins

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
When compiling the test plugins, we now run `go build` from within the server directory and force the usage of the vendored dependencies. This effectively couples the test plugins dependencies with the servers dependencies.

Requires https://github.com/mattermost/mattermost-server/pull/12239

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-18193

### CODE REVIEW

**Comment by hanzei** [comment]
Yes, you are correct @streamer45, I have to fix this CI. Requested the reviews before waiting for it to run.

**Comment by hanzei** [comment]
Using go modules all the time will be done with https://github.com/mattermost/mattermost-server/pull/12040. Hence, I will pause this PR, until https://github.com/mattermost/mattermost-server/pull/12040 is merged.

**Comment by amyblais** [comment]
Removed from v5.16 because it didn't make it to feature complete. Please let me know if this is important for v5.16.

**Comment by hanzei** [comment]
I would like to deffer this question to @lieut-data. At the core this is a fix for flaky tests, which would be handy to have in the release branch.

**Comment by lieut-data** [comment]
@amyblais, we need this in v5.16 to fix a long chain of issues that prevent consistently successful release building. While it's marked as a feature, it's really just an action item in resolving a larger bug.

**Comment by amyblais** [comment]
Sounds good,

**Comment by hanzei** [comment]
@lieut-data @streamer45 This is ready for a final review. Due to a bug in go (https://github.com/golang/go/issues/30374) the go patch version had to be bumped. Because of this, I based this PR on top of https://github.com/mattermost/mattermost-server/pull/12239.

File: scripts/test.sh [hunk 1]
```diff
@@ -15,7 +15,7 @@ echo "Packages to test: $PACKAGES"
 find . -name 'cprofile*.out' -exec sh -c 'rm "{}"' \;
 find . -type d -name data -not -path './vendor/*' | xargs rm -rf
 
-$GO test $GOFLAGS -run=$TESTS $TESTFLAGS -v -timeout=2000s -covermode=count -coverpkg=$PACKAGES_COMMA -exec $DIR/test-xprog.sh $PACKAGES 2>&1 | tee output
+GO111MODULE=on $GO test -mod=vendor $GOFLAGS -run=$TESTS $TESTFLAGS -v -timeout=2000s -covermode=count -coverpkg=$PACKAGES_COMMA -exec $DIR/test-xprog.sh $PACKAGES 2>&1 | tee output
```
Comment (hanzei) [inline scripts/test.sh]: This is just a short term fix. A more holistic fix will be includes in https://github.com/mattermost/mattermost-server/pull/12040. I will rip this out, when the other PR gets merged.

**Comment by hanzei** [comment]
@crspeller @cpanato Could one of you please review this given given that Jesse is OOO?

**Comment by lieut-data** [comment]
@prapti, there isn't anything specific to this change to be tested: the Go code changed only affects the unit tests themselves. However, we are upgrading the Go build tooling as part of this, which technically affects everything -- so ideally we just merge this and let it soak on community.

**Comment by prapti** [comment]
@lieut-data Sounds good! I'll go ahead and approve it.

**Pull request merged**